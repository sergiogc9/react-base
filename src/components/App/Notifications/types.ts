import { Notification, QueueNotification } from '@src/types/notification';

export type StateProps = {
	notifications: QueueNotification[]
};

export type DispatchProps = {
	onAddNotification: (notification: Notification) => void,
	onUnqueueNotification: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps;
