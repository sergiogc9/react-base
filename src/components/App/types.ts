import { TProps } from '@src/lib/i18n';

export type StateProps = {
	authenticated: boolean
};

export type DispatchProps = {
	onAuth: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
