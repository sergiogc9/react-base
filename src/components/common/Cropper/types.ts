type Props = {
	/**
	 * The image in Base64. If null, no image has been set before.
	 */
	readonly image: string;
	/**
	 * A callback method called with the cropped image data
	 */
	readonly onImageCropped: (image: string) => void;
};

export type CropperProps = Props;
