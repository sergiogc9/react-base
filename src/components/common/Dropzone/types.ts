import { TProps } from '@src/lib/i18n';

export type AddFileError = {
	error: 'file_too_large' | 'incorrect_file_type'
	file: File
};

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	files: File[]
	multiple: boolean
	accept?: string
	maxSize?: number
	showError?: boolean
	onFilesAdded(files: File[]): void
	onAddFilesError(errors: AddFileError[]): void
	onRemoveFile(file: File): void
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
