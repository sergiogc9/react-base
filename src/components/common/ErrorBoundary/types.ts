export type ErrorBoundaryProps = React.PropsWithChildren;
