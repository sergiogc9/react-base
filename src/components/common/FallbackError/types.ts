import { FallbackProps } from 'react-error-boundary';

export type FallbackErrorProps = FallbackProps;
