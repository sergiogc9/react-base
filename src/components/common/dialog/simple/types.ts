import { TProps } from '@src/lib/i18n';

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	id: string
	text: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
