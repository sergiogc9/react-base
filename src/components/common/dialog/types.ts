import { TProps } from '@src/lib/i18n';

export type StateProps = {
};

export type DispatchProps = {
};

export type OwnProps = {
	id: string
	icon?: JSX.Element
	title: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
