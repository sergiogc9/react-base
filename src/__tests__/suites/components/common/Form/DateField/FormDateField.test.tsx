import React from "react";
import { DatePicker } from "react-md/lib/Pickers";
import { act } from 'react-dom/test-utils';
import moment from 'moment';
import mockDate from 'mockdate';

import FormDateField from "@src/components/common/Form/DateField/";
import { ComponentProps } from "@src/components/common/Form/DateField/types";
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const element = TestHelper.dateFieldElement;
const user = TestHelper.getUser();
const date = new Date();
const expectedDate = moment.tz(moment(date).format('YYYY-MM-DD'), "UTC").toDate();

let onChangeMock = jest.fn();
describe("Date field common component", () => {
	let wrapper: any;

	const updateComponent = (props: Partial<ComponentProps> = {}) => {
		act(() => {
			wrapper = TestHelper.getWrappedComponent(
				<FormDateField
					element={element}
					error={false}
					onChangeDate={onChangeMock}
					{...props}
				/>, { app: { profile: { user } } }).component;
		});
	};

	beforeAll(() => {
		mockDate.set(new Date("2019-10-28"));
	});

	afterAll(() => {
		mockDate.reset();
	});

	beforeEach(() => {
		onChangeMock = jest.fn();
		updateComponent();
	});

	it("Component default snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component on change date input", () => {
		act(() => wrapper.find(DatePicker).at(0).props().onChange(null, new Date(), null));
		expect(onChangeMock).toHaveBeenCalledTimes(1);
	});

	it("Component on validate base validation", () => {
		updateComponent({ element: { ...element, validations: { fn: () => false } } });
		act(() => wrapper.find(DatePicker).at(0).props().onChange(null, date, null));
		expect(onChangeMock).toHaveBeenCalledWith(element.id, expectedDate, false);
	});

	it("Component on validate correct", () => {
		updateComponent({ element: { ...element, validations: {} } });
		act(() => wrapper.find(DatePicker).at(0).props().onChange(null, date, null));
		expect(onChangeMock).toHaveBeenCalledWith(element.id, expectedDate, true);
	});

	it("Component snapshot after on force update", () => {
		updateComponent({ forceValue: new Date(0) });
		expect(onChangeMock).toHaveBeenCalledWith(element.id, new Date(0), true);
		expect(wrapper.html()).toMatchSnapshot();
	});
});
