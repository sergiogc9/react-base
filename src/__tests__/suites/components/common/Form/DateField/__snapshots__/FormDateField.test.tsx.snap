// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date field common component Component default snapshot 1`] = `"<div class=\\"react-base-form-date-field\\"><div class=\\"md-picker-container\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input md-pointer--hover\\"><label for=\\"formDateField-dateFieldId\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">insert_article.modal.form.date</label><input readonly=\\"\\" id=\\"formDateField-dateFieldId\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text md-pointer--hover\\" value=\\"10/28/2019\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div></div>"`;

exports[`Date field common component Component snapshot after on force update 1`] = `"<div class=\\"react-base-form-date-field\\"><div class=\\"md-picker-container\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input md-pointer--hover\\"><label for=\\"formDateField-dateFieldId\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">insert_article.modal.form.date</label><input readonly=\\"\\" id=\\"formDateField-dateFieldId\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text md-pointer--hover\\" value=\\"1/1/1970\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div></div>"`;
